api/src/main/java/org/openmrs/module/patientqueueing/api/PatientQueueingService.java

```diff
@@ -9,12 +9,76 @@
  */
 package org.openmrs.module.patientqueueing.api;
 
+import org.openmrs.Location;
+import org.openmrs.Patient;
+import org.openmrs.Provider;
+import org.openmrs.api.APIException;
 import org.openmrs.api.OpenmrsService;
+import org.openmrs.module.patientqueueing.model.PatientQueue;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Date;
+import java.util.List;
 
 /**
  * The main service of this module, which is exposed for other modules. See
  * moduleApplicationContext.xml on how it is wired up.
  */
 public interface PatientQueueingService extends OpenmrsService {
 	
+	/**
+	 * Get a single patient queue record by queueId. The queueId can not be null
+	 * 
+	 * @param queueId Id of the PatientQueue to be retrieved
+	 * @return The Patient Queue that matches the queueId
+	 * @throws APIException
+	 */
+	@Transactional(readOnly = true)
+	public PatientQueue getPatientQueueById(Integer queueId) throws APIException;
+	
+	/**
+	 * Update or Save patientQueue. Requires a patientQueue
+	 * 
+	 * @param patientQueue The PatientQueue to be saved
+	 * @return PatientQueue that has been saved
+	 * @throws APIException
+	 */
+	@Transactional
+	public PatientQueue savePatientQue(PatientQueue patientQueue) throws Exception;
+	
+	/**
+	 * Gets a list of patient queues basing on given parameters.
+	 * 
+	 * @param provider The provider where the patient was being sent. It Can be null
+	 * @param fromDate lowest date a query will be built upon. It can be null
+	 * @param toDate highest date a query will be built upon. It Can be null
+	 * @param locationTo Location Where patient was sent to
+	 * @param locationFrom Location Where patient was sent from
+	 * @param patient The patient who is in the queue
+	 * @param status Status such as COMPLETED,PENDING
+	 * @return List<PatientQueue> A list of patientQueue that meet the parameters
+	 * @throws APIException
+	 */
+	@Transactional(readOnly = true)
+	public List<PatientQueue> getPatientQueueList(Provider provider, Date fromDate, Date toDate, Location locationTo,
+	        Location locationFrom, Patient patient, String status) throws APIException;
+	
+	/**
+	 * Mark passed patientQueue completed.
+	 * 
+	 * @param patientQueue The PatientQueue to be completed
+	 * @return PatientQueue. The Queue that is completed
+	 * @throws APIException
+	 */
+	@Transactional
+	public PatientQueue completePatientQueue(PatientQueue patientQueue) throws Exception;
+	
+	/**
+	 * This method gets the patientQueue for a patient at a given location which is not complete.
+	 * @param locationTo The Location where the patient was is queued to
+	 * @param patient The patient who is in the queue
+	 * @return a patient queue that meets the criteria of parameters
+	 */
+	@Transactional(readOnly = true)
+	public PatientQueue getPatientQueue(Patient patient, Location locationTo);
```

Comment:
If this is fetching a patient queue which is not complete, then it should be reflected in the name.